src/duckling/project/snackbar.service.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+import {MdSnackBar} from '@angular/material';
+
+export interface SnackBar {
+    message: string,
+    action: string
+}
+
+/**
+ * Service used to display snackbars one after the other
+ */
+@Injectable()
+export class SnackBarService {
+    private _snacks : SnackBar[] = [];
+
+    constructor(private _snackbar : MdSnackBar) {
+    }
+
+    /**
+     * Invoke all the current snackbars, first one put in is the first one displayed
+     */
+    invokeSnacks() {
+        if (this._snacks.length > 0) {
+            let ref = this._snackbar.open(this._snacks[0].message, this._snacks[0].action);
+            this._snacks = this._snacks.slice(1);
+            ref.afterDismissed().subscribe(() => this.invokeSnacks());
+        }
+    }
+
+    /**
+     * Adds a new snackbar to the service
+     */
+    addSnack(snack : SnackBar) {
```

Comment:
I don't think the API should require the consumer to build a snackbar. It should accept a string message. We can expose this as a separate API as needed in the future.